modules/javafx.graphics/src/main/java/javafx/scene/paint/Stop.java

```diff
@@ -184,7 +264,44 @@ public final Color getColor() {
      */
     public Stop(@NamedArg("offset") double offset, @NamedArg(value="color", defaultValue="BLACK") Color color) {
         this.offset = offset;
-        this.color = color;
+        this.color = Objects.requireNonNullElse(color, Color.TRANSPARENT);
```

Comment:
Note that a `null` color is now treated as `TRANSPARENT` for the following reasons:
1. The previous implementation was broken: if a `Stop` is constructed with a `null` color, the `hashCode()` method throws a NPE because it doesn't check for `null`.
2. It doesn't make sense to have a stop with `null` color. What does it even mean?
3. When the stop list is normalized, an empty or null list is treated as a two-stop list with `TRANSPARENT` color (see `normalize()`). So we already have a scenario where `null` is treated as `TRANSPARENT`.